src/parser/parse_tree_cl.c

```diff
@@ -15993,10 +15994,13 @@ pt_print_value (PARSER_CONTEXT * parser, PT_NODE * p)
     case PT_TYPE_DATETIME:
     case PT_TYPE_DATETIMETZ:
     case PT_TYPE_DATETIMELTZ:
-      if (p->info.value.text)
+      if (!(parser->custom_print & PT_PRINT_SUPPRESS_FOR_DBLINK))
 	{
-	  q = pt_append_nulstring (parser, q, p->info.value.text);
-	  break;
+	  if (p->info.value.text)
```

Comment:
It's a good idea to check the "p->info.value.text" condition first.
The same goes for the code below.

Reply:
I think the print_custom comparison should come first. Because value.text compares with another comparison.